### ISSUE TITLE
Livewire

### ISSUE THREAD

**Issue by i-prikot** [WendellAdriel/laravel-validated-dto#88]
Hi.

How to use Wireable with Livewire. Can you write a usage example? Can't call validation in a livewire component method?

DTO must inherit interface Livewire\Wireable.

**Comment by i-prikot** [comment]
Hi. Of course, here's an example.

```
namespace App\Livewire\Components;

use Livewire\Component;

class ProjectName extends Component
{
   public ExampleDTO $dto;
 
   public function mount(ExampleDTO $dto)
    {
        // Here error. ValidatedDTO call validate
        $this->dto = $dto;
    }

   public function save()
   {
       //
   }
}
```

It should look something like this, but it doesn't work either.

```
namespace App\DTOs\Projects;

use WendellAdriel\ValidatedDTO\Concerns\Wireable;
use Livewire\Wireable;

class ExampleDTO extends ValidatedDTO implements Wireable
{
    use Wireable;
    ...
}
```
```
<form wire:submit="save">
   <input type="text" wire:model="dto.name" />
</form>
```
Livewire has its own interface Livewire\Wireable.
https://livewire.laravel.com/docs/properties#supporting-custom-types

**Comment by WendellAdriel** [comment]
Oh ok, now I see what you mean.
Since you have it in the mount, it will try to instantiate the object, but then the validation will fail.

The thing I have in mind, but I'll need to test an implementation around it, is creating a way to make a Lazy Validation, so it will validate only when you call a `->validate()` method for example.

**Comment by WendellAdriel** [comment]
Hey @i-prikot,
I just released v3.10.0 with a lazy validation feature on it.
The feature is explained here: https://wendell-adriel.gitbook.io/laravel-validated-dto/basics/lazy-validation

Can you check if with this you're able to better use the DTO in Livewire now?

**Comment by i-prikot** [comment]
> Hey @i-prikot, I just released v3.10.0 with a lazy validation feature on it. The feature is explained here: https://wendell-adriel.gitbook.io/laravel-validated-dto/basics/lazy-validation
> 
> Can you check if with this you're able to better use the DTO in Livewire now?

Hey @WendellAdriel 

Now I can use DTO in Livewire. I really appreciate all the time and effort you've put into this. Great job!

**Comment by WendellAdriel** [comment]
Oh, that's great to hear @i-prikot!!! 🥳 
I'm happy to help, thanks for the heads up on this issue! 💪 
I hope this library can help you on your projects, have a great day over there!
